lib/internal/url.js

```diff
@@ -239,8 +239,10 @@ class URL {
   constructor(input, base) {
     // toUSVString is not needed.
     input = `${input}`;
-    if (base !== undefined && !(base instanceof URL))
+    if (base !== undefined &&
+        (!base[searchParams] || !base[searchParams][searchParams])) {
```

Comment:
Performance-wise? No. There's actually a small but measurable performance hit when extracting all of the logic to a separate function, compared to having it inline. For example:

```
                                                                              improvement confidence      p.value
 url/url-searchparams-read.js n=20000000 param="nonexistent" method="get"       165.07 %        *** 1.189398e-66
 url/url-searchparams-read.js n=20000000 param="nonexistent" method="getAll"    139.46 %        *** 4.331942e-46
 url/url-searchparams-read.js n=20000000 param="nonexistent" method="has"       161.48 %        *** 2.741360e-52
 url/url-searchparams-read.js n=20000000 param="one" method="get"               222.70 %        *** 3.111318e-46
 url/url-searchparams-read.js n=20000000 param="one" method="getAll"            113.31 %        *** 1.122968e-53
 url/url-searchparams-read.js n=20000000 param="one" method="has"               232.04 %        *** 1.571222e-55
 url/url-searchparams-read.js n=20000000 param="three" method="get"             184.13 %        *** 2.078669e-48
 url/url-searchparams-read.js n=20000000 param="three" method="getAll"          100.34 %        *** 4.591714e-56
 url/url-searchparams-read.js n=20000000 param="three" method="has"             181.29 %        *** 7.340053e-43
 url/url-searchparams-read.js n=20000000 param="two" method="get"               209.77 %        *** 4.581349e-54
 url/url-searchparams-read.js n=20000000 param="two" method="getAll"            102.89 %        *** 2.001755e-36
 url/url-searchparams-read.js n=20000000 param="two" method="has"               203.09 %        *** 1.810527e-44
```